### PR TITLE
Rename financial field

### PR DESCRIPTION
**Pull request by akainic**
Rename financial field to `discloseFinancialInformation` and change enrollment-system.js to include the financial fields if `discloseFinancialInfo` is true instead of being dependent on whether incomes and expenses exist. I'm not sure if this change makes sense, but since the front-end should be guaranteeting that incomes and expenses exist if they select "yes" for `discloseFinancialInformation`, it should always be the case.

Still working through some conformance test issues, but wanted to push up for review.

### CODE REVIEW

**Comment by akainic** [comment]
Tagging @alexose and @markolson (can wait until Monday) for review.

File: src/server/enrollment-system.js
Comment (markolson) [inline src/server/enrollment-system.js]: Based on my reading of the PR description, I think we want to just do something like
```
if (!veteran.discloseFinancialInformation) { 
    return undefined;
}
```
here so that `marriedLastCalendarYear` and `numberOfDependentChildren` aren't forcing `financialStatement` to be present if `discloseFinancialInformation` is false.

File: src/server/enrollment-system.js
Comment (markolson) [inline src/server/enrollment-system.js]: At this point, we should probably change this line to be `incomeTest: { discloseFinancialInformation: true }` because it won't be included either way if `discloseFinancials` is false. Let's just remove the indirection..

File: test/data/conformance/no-financial.xml
Comment (markolson) [inline test/data/conformance/no-financial.xml]: `discloseFinancialInformation` is false, but `financialsInfo` is still present here. This is the test I sort of want to verify is behaving correctly, as referenced in my other comment.

**Review by markolson — CHANGES_REQUESTED** [review]
I think this is 90% of the way there, but want to verify (and change, if needed) some of the logic. Changes to the test data looks good and thorough!

File: src/server/enrollment-system.js
Comment (akainic) [inline src/server/enrollment-system.js]: Yep that makes sense!

File: src/server/enrollment-system.js
Comment (akainic) [inline src/server/enrollment-system.js]: Ah, so you mean if I make the change you identified above, with returning undefined if `!veteran.discloseFinancialInformation`, then we assume it will be true if it reaches this point?

File: test/data/conformance/no-financial.xml
Comment (akainic) [inline test/data/conformance/no-financial.xml]: Yeah I was very confused about how to fix the conformance tests, this might be included because of the logic changes you identified earlier, so once I make those this should go away, I just copied it in based on wha the failing test expected to see.

File: src/server/enrollment-system.js
Comment (markolson) [inline src/server/enrollment-system.js]: Yep!

File: test/data/conformance/no-financial.xml
Comment (markolson) [inline test/data/conformance/no-financial.xml]: Yep, that's exactly how I did it - these conformance tests are totally supposed to "catch" these sort of logic changes, and let you take a step back and look at how different applications change in response to things like the above change in enrollment-systems.

**Comment by akainic** [comment]
@markolson made the changes you requested! Still working on the marital status change (might include that in a separate PR as this one is already large). Let me know how this looks to you so far.

**Review by markolson — APPROVED** [review]

**Comment by markolson** [comment]
LGTM! I'll do another review when the marital changes come in

**Comment by akainic** [comment]
@markolson this is ready for your review again!

**Comment by alexose** [comment]
LGTM, again :)

**Review by markolson — APPROVED** [review]

**Pull request merged**
